frontend/src/Components/ProductListingPage/MobileSortAndFilter.jsx

```diff
@@ -0,0 +1,134 @@
+
+import { useProduct } from "../../Context/ProductContext";
+
+import "./MobileSortAndFilter.css";
+import "./FilterSideBar.css"
+const sortByNames = ["Latest","Discount","Price : High to Low","Price : Low to High"]
+export const MobileSortAndFilter = () => {
+
+
+    const {state,dispatch} = useProduct();
+
+    const sortButtonHandler = () => {
+        dispatch({type:"OPEN_SORT"})
+        dispatch({type:"SET_OVERLAY"})
+    }
+
+    const filterButtonHandler = () => {
+        dispatch({type:"OPEN_FILTER"})
+        dispatch({type:"SET_OVERLAY"})
+    }
+
+
+    const closeFilterHandler = () => {
+        dispatch({type:"CLEAR_ALL_FILTERS"});
+        dispatch({type:"OPEN_FILTER"});
+        dispatch({type:"SET_OVERLAY"})
+
+
+    }
+
+    const getBrands = (products) => {
```

Comment:
You can probably extract this out in utils to use it in both desktop and mobile filter files

Reply:
done